### PR TITLE
Solve factorial TDD

### PR DESCRIPTION
**Pull request by tonnymuchui**
- Create a class called Solver :heavy_check_mark: 
- Created a method called factorial :heavy_check_mark: 
- Create a method called reverse :heavy_check_mark: 
- Create a method called fizz buzz :heavy_check_mark:

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      'fizzbuzz'
+    elsif (number % 3).zero?
+      'fizz'
+    elsif (number % 5).zero?
+      'buzz'
+    else
+      number.to_s
+    end
+  end
+end
```
Comment (Ismailco) [inline solver.rb:27]: - [ ] This is a TDD project, so the commit history sequence should clearly show that each of the Solver class methods was tested before writing the code, kindly make sure to write your tests before implementing the Code

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @tonnymuchui, @Fahim-Khan-P, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ Used Rspec for Unit-Test
✔️ Tests working properly
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: solver.rb (hunk 1 above)
Comment (Fahim-Khan-P) [reply solver.rb:27]: Hi @Ismailco,
Thank you for giving us time to share your beneficial suggestion, We will consider it next time.

**Review by DammyShittu — APPROVED** [review]
Hi @tonnymuchui and @Fahim-Khan-P,

## APPROVED 🟢

## Highlights
✔️ TDD for testing

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Fahim-Khan-P** [comment]
Thank you, dear @tonnymuchui, for giving us your expensive time!

**Pull request merged**